### PR TITLE
Fix for Orb of Confinement in Achaea #77

### PR DESCRIPTION
**Pull request by PapaGuacamole**
Echoing twice when the variable changed by trigger:

```
You say, "Duanatharan."
A shimmering orb covers the city, preventing you from rising to the skies.
(mapper): Orb of Confinement registered as active for Ashtan

(mapper): Orb of Confinement registered as active for Ashtan
14988h|150%, 8651m|99% ex|cdbk-80%  14:36:49.239
```

### CODE REVIEW

**Comment by PapaGuacamole** [comment]
Still echoing twice, can't work out why. Also not remapping path when autodetecting orb is active

**Comment by PapaGuacamole** [comment]
OK, this is now done, testing and working

File: mudlet-mapper.xml [hunk 1]
```diff
@@ -5872,7 +5940,7 @@ function mmp.addWingsAchaea()
 		if
     	mmp.settings.duanatharic and
     	gmcp.Room and
-    	not table.contains(gmcp.Room.Info.details, &quot;indoors&quot;)
+    	not table.contains(gmcp.Room.Info.details, &quot;indoors&quot;) and not orbed
```
Comment (vadi2) [inline mudlet-mapper.xml]: Do you mean `orbed()`?

File: mudlet-mapper.xml [hunk 2]
```diff
@@ -5899,7 +5967,7 @@ function mmp.addWingsAchaea()
 		
 	-- Air Elemental Aero Soar
 	  if
-			mmp.settings.soar and gmcp.Room and not table.contains(gmcp.Room.Info.details, &quot;indoors&quot;)
+			mmp.settings.soar and gmcp.Room and not table.contains(gmcp.Room.Info.details, &quot;indoors&quot;) and not orbed
```
Comment (vadi2) [inline mudlet-mapper.xml]: Do you mean `orbed()`?

**Comment by PapaGuacamole** [comment]
Yes, yes I did. Now updated

File: mudlet-mapper.xml [hunk 3]
```diff
@@ -5834,7 +5891,17 @@ function mmp.addWingsAchaea()
...
-
+	
+	local function orbed()-- orb of confinement - PapaGuacamole
+		return
+				(getRoomArea(mmp.currentroom) == 49 and mmp.settings.orbashtan) or
+    		(getRoomArea(mmp.currentroom) == 57 and mmp.settings.orbcyrene) or
+    		(getRoomArea(mmp.currentroom) == 51 and mmp.settings.orbeleusis) or
+    		(getRoomArea(mmp.currentroom) == 55 and mmp.settings.orbhashan) or
+    		(getRoomArea(mmp.currentroom) == 44 and mmp.settings.orbmhaldor) or
+    		(getRoomArea(mmp.currentroom) ==368 and mmp.settings.orbtargossas)
```
Comment (keneanung) [inline mudlet-mapper.xml]: Maybe call `getRoomArea(mmp.currentroom))` only once? It shouldn't change between those calls :stuck_out_tongue:

Comment (PapaGuacamole) [reply mudlet-mapper.xml]: I have updated that in pull request 81, this one's kind of been superseded by that one (because I forked at the wrong point)

Comment (keneanung) [reply mudlet-mapper.xml]: I can't see the change there either. Did you push your branch?

Comment (PapaGuacamole) [reply mudlet-mapper.xml]: I ended up with two similar branches, with the changes not always going to the right one! https://github.com/IRE-Mudlet-Mapping/ire-mapping-script/pull/82 is the right one

**Pull request merged**